### ISSUE TITLE
Error Activating AWS Toolkit

### ISSUE THREAD

**Issue by digitaltoast** [aws/aws-toolkit-vscode#1229]
**Describe the bug**

At some point in the last 2 weeks, "AWS Toolkit Output" log window started showing:
"Error Activating AWS Toolkit"

**To Reproduce**

N/A

**Expected behavior**

AWS Toolkit logs in

**Desktop (please complete the following information):**

OS:  Linux x64 5.4.0-1018-aws
Visual Studio Code Version:  1.47.3
AWS Toolkit Version:  1.12.0

**Additional context**

All the icons next to AWS: Explorer respond with "not found" (see screenshot below).

I have tried uninstalling and re-installing the extension in the remote explorer. I have also tried re-logging in with a different .aws profile. 

```
Error logs for this session are permanently stored in /home/ubuntu/.config/Code/logs/aws_toolkit/aws_toolkit_20200805T111618.log
2020-08-05 11:16:19 [INFO]: log level: info
2020-08-05 11:16:19 [INFO]: Retrieving AWS endpoint data
2020-08-05 11:16:19 [INFO]: OS:  Linux x64 5.4.0-1018-aws
Visual Studio Code Version:  1.47.3
AWS Toolkit Version:  1.12.0

2020-08-05 11:16:23 [ERROR]: spawn /home/ubuntu/.vscode-server/bin/91899dcef7b8110878ea59626991a18c8a6a1b3e/node_modules/vscode-ripgrep/bin/rg EACCES
2020-08-05 11:21:20 [INFO]: Successfully sent a telemetry batch of 2
```

![image](https://user-images.githubusercontent.com/230370/89402228-6e675080-d70e-11ea-8769-8826291e8921.png)

**Comment by justinmk3** [comment]
Does this happen every time (i.e. it doesn't sometimes work and sometimes not)?

Does it happen in these older Toolkit versions ? 

- https://github.com/aws/aws-toolkit-vscode/releases/download/v1.10.0/aws-toolkit-vscode-1.10.0.vsix
- https://github.com/aws/aws-toolkit-vscode/releases/download/v1.11.0/aws-toolkit-vscode-1.11.0.vsix

To install each `.vsix` , 

1. in VS Code, go to the command palette (ctrl-shift-p or cmd-shift-p)
2. Run the `Install from VSIX` command
3. find the `.vsix` file you downloaded
4. click the "Reload" button
5. Ensure that version 1.10 (or 1.11) is installed--don't "update to 1.12" :)

**Comment by justinmk3** [comment]
https://github.com/aws/aws-toolkit-vscode/pull/1241 hopefully fixes this, that will be included in the next release.

Can you try it to confirm? [aws-toolkit-vsix.zip](https://github.com/aws/aws-toolkit-vscode/files/5086582/aws-toolkit-vsix.zip)

1. Unzip `aws-toolkit-vsix.zip`
2. In VSCode run the  `Install from VSIX...` to install from the `.vsix` file
3. Reload VSCode and check that the Toolkit has the new version

**Comment by sliedig** [comment]
I'm having the same issue. the VSIX you shared did not fix it. 

VS Code Info:
Version: 1.48.0
Commit: db40434f562994116e5b21c24015a2e40b2504e6
Date: 2020-08-13T08:03:27.059Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.6.0

**Comment by justinmk3** [comment]
@sliedig thanks! In vscode, when this error occurs, there should still be a `AWS: View AWS Toolkit Logs` command available from the vscode command palette (ctrl-shift-p or cmd-shift-p). Can you share the contents of that log file?

Also the `Log (Extension Host)` output may be helpful (available in VSCode from the **Output** tab, then select `Log (Extension Host)` in the menu).

**Comment by sliedig** [comment]
Wow, that is weird. It's working all-of-a-sudden. I have no idea what happened, but it works now.

**Comment by sliedig** [comment]
Stike the last comment. I am still experiencing the issue, but think I have narrowed the problem down. Seems that if you have more than one window open the extension doesn't load properly. There are also no output for logs and I am not using the remote feature. 

Here is a screenshot of two windows. As you can see the AWS plugin is activated on the back window, not the front.

![image](https://user-images.githubusercontent.com/597073/90719723-bb8b1b00-e2e7-11ea-9f2a-293717f77df1.png)

**Comment by justinmk3** [comment]
@sliedig can you check `Log (Extension Host)` ? Are you sure Toolkit is version 1.13 (not 1.12)?

> Seems that if you have more than one window open the extension doesn't load properly

That definitely sounds related to the global user settings store....

**Comment by sliedig** [comment]
[2020-08-21 08:58:55.204] [exthost] [info] ExtensionService#_doActivateExtension amazonwebservices.aws-toolkit-vscode {"startup":false,"extensionId":{"value":"amazonwebservices.aws-toolkit-vscode","_lower":"amazonwebservices.aws-toolkit-vscode"},"activationEvent":"onStartupFinished"}
[2020-08-21 08:58:55.204] [exthost] [info] ExtensionService#loadCommonJSModule file:///Users/sliedig/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.12.0/extensionMain
[2020-08-21 08:58:55.465] [exthost] [error] Activating extension amazonwebservices.aws-toolkit-vscode failed due to an error:
[2020-08-21 08:58:55.466] [exthost] [error] SecurityError: localStorage is not available for opaque origins

**Comment by justinmk3** [comment]
>  AWS Toolkit Version: 1.12.0

@sliedig It should be 1.13 if testing the vsix given in https://github.com/aws/aws-toolkit-vscode/issues/1229#issuecomment-675093923

> [2020-08-21 08:58:55.465] [exthost] [error] Activating extension amazonwebservices.aws-toolkit-vscode failed due to an error:
> [2020-08-21 08:58:55.466] [exthost] [error] SecurityError: localStorage is not available for opaque origins

Ok, that at least confirms that this is related to localStorage, which very much helps. Can you confirm that the same error happens with version 1.13 ?

**Comment by bryceitoc9** [comment]
This was released on August 24th in the toolkit's 1.13.0 release.

**Comment by justinmk3** [comment]
If  this issue persists with version 1.13 of AWS Toolkit, please let us know here.  We may need to update more parts of the toolkit that access vscode "storage": from https://github.com/aws/aws-toolkit-vscode/issues/1229#issuecomment-677986538 : 

> TODO: There are various places referencing `globalStorage`, [globalState](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext), [globalStoragePath](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext), which may need to be updated to handle this error case.
